site/source/docs/migrating_from_3.x.md

```diff
@@ -83,25 +83,144 @@ However, we did introduce some small breaking changes. We removed:
+            
+Here is an example of how simple your blockchain config can look now:
+
+```javascript
+module.exports = {
+  default: {
+    enabled: true,
+    client: "geth"
+  },
+
+  development: {
+    endpoint: 'ws://locahost:1234',
+    clientConfig: {
+      miningMode: 'dev'
+    },
+    accounts: [
+      {
+        mnemonic: "12 word mnemonic",
+        balance: "5ether",
+        numAddresses: 10
+      }
+    ]
+  }
+}
+```
+
+This will connect to a node started on you machine on the port `1234`, using a Websocket provider. If the node is not started, Embark will start it for you using Geth.
+
+The accounts used for deployment will be the ones coming from a mnemonic, all funded with 5 ETH.
+
+You can even use those accounts in your Dapp to test your contract interactions with `$EMBARK` set in your `dappConnection`, in the contracts config.
+
+You could remove the `endpoint` and `accounts` and Embark would still work with its defaults.
+
+### Tests
+
+There were few breaking changes in the test, mostly new features.
+
+1. The `deployment` section was removed
+    - Much like the changes above, the `deployment` section is no more
+    - You can still connect to an extenral node in tests and use custom accounts, it's just simpler
+    - You can now add a `blockchain` section in the `config` function's object and use the `endpoint` and `accounts` options
+2. `contracts` was renamed `deploy`
+    - Again, the test configs match the actual configs
+    - You now have the `contracts` section which represents the contracts config, so you need the `deploy` parameter for the contract list
+    - See the example below
+3. New module configs
+    - You can now configure the namesystem on a per test basis
+    - Storage, Namesystem and Communication modules are disabled by default to speed up the test
+        - You can enable them in your normal config files by adding the `test` environment and putting `enabled: true`
+            - Just like normal environments, the `test` environment merges with the `default` section
+            - All tests now default to the `test` environment, but you can still change this with `--environment`
+            
+Example of a test with the new configs:
+
+```javascript
+config({
+  blockchain: {
+    accounts: [
+      {
+        mnemonic: "12 word mnemonic",
+        balance: "5ether",
+        numAddresses: 10
+      }
+    ]
+  },
+  contracts: {
+    deploy: {
+      "SimpleStorage": {
+        args: [100]
+      }
+    }
+  },
+  namesystem: {
+    enabled: true,
+    register: {
+      "rootDomain": "test.eth"
+    }
+  }
+});
+```
+
+This example deploys one contract, SimpleStorage, using the first of 10 accounts, generated from the mnemonic, which will be funded to 5 ETH.
+
+There will also be the Namesystem enabled, with a root domain of `test.eth`, meaning you will be able, in your tests, to do `EmbarkJS.Names.resolve('test.eth')` and it will resolve to the owner of the name (the default account).
```

Comment:
```suggestion
The Namesystem will also be enabled, with a root domain of `test.eth`, meaning you will be able, in your tests, to do `EmbarkJS.Names.resolve('test.eth')` and it will resolve to the owner of the name (the default account).
```

Here and elsewhere, not sure about capitalizing *[N/n]amesystem*.